examples/redis-plugin/main.go

```diff
@@ -25,65 +21,9 @@ func main() {
 	core.EventLoopWithInterrupt(agent, exampleFinished)
 }
 
-// ExampleFlavor is a set of plugins required for the redis example.
-type ExampleFlavor struct {
-	// Local flavor to access to Infra (logger, service label, status check)
-	*local.FlavorLocal
-
-	// Redis plugin
-	Redis         redis.Plugin
-	RedisDataSync kvdbsync.Plugin
-
-	ResyncOrch resync.Plugin
-
-	// Example plugin
-	RedisExample ExamplePlugin
-
-	// For example purposes, use channel when the example is finished
-	closeChan *chan struct{}
-}
-
-// Inject sets object references
-func (ef *ExampleFlavor) Inject() (allReadyInjected bool) {
-	// Init local flavor
-	if ef.FlavorLocal == nil {
-		ef.FlavorLocal = &local.FlavorLocal{}
-	}
-	ef.FlavorLocal.Inject()
-	ef.Redis.Deps.PluginInfraDeps = *ef.InfraDeps("redis",
-		local.WithConf())
-	InjectKVDBSync(&ef.RedisDataSync, &ef.Redis, ef.Redis.PluginName, ef.FlavorLocal, &ef.ResyncOrch)
-	ef.RedisExample.Deps.PluginLogDeps = *ef.FlavorLocal.LogDeps("redis-example")
-	ef.RedisExample.closeChannel = ef.closeChan
-
-	return true
-}
-
-// InjectKVDBSync helper to set object references
-func InjectKVDBSync(dbsync *kvdbsync.Plugin,
-	db keyval.KvProtoPlugin, dbPlugName core.PluginName, local *local.FlavorLocal, resync resync.Subscriber) {
-
-	dbsync.Deps.PluginLogDeps = *local.LogDeps(string(dbPlugName) + "-datasync")
-	dbsync.KvPlugin = db
-	dbsync.ResyncOrch = resync
-	if local != nil {
-		dbsync.ServiceLabel = &local.ServiceLabel
-
-		if local.StatusCheck.Transport == nil {
-			local.StatusCheck.Transport = dbsync
-		}
-	}
-}
-
-// Plugins combines all Plugins in flavor to the list
-func (ef *ExampleFlavor) Plugins() []*core.NamedPlugin {
-	ef.Inject()
-	return core.ListPluginsInFlavor(ef)
-}
-
-// Deps is a helper struct which is grouping all dependencies injected to the plugin
-type Deps struct {
-	local.PluginLogDeps // injected
+// Redis flag to load config
+func init() {
+	flag.String("redis-config", "redis.conf", "Location of the redis configuration file")
```

Comment:
Latest cn-infra allows to avoid copy&paste flags for plugin configs (see WithConf() option).

InfraPluginDeps("", WithConf())

Reply:
done